packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+  /**
+   * Additional configuration for FSx specific to Lustre.
+   */
+  readonly lustreConfiguration: LustreConfiguration;
+
+  /**
+   * The subnet that the file system will be accessible from.
+   */
+  readonly vpcSubnet: ISubnet;
+}
+
+/**
+ * The FSx for Lustre File System implementation of IFsxFileSystem.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-fsx-filesystem.html
+ *
+ * @resource AWS::FSx::FileSystem
+ */
+export class FsxLustreFileSystem extends FsxFileSystemBase {
+
+  /**
+   * Import an existing FSx for Lustre file system from the given properties.
+   */
+  public static fromFsxLustreFileSystemAttributes(scope: Construct, id: string, attrs: FsxFileSystemAttributes): IFsxFileSystem {
+    class Import extends FsxFileSystemBase {
+      public readonly fileSystemId = attrs.fileSystemId;
+      public readonly connections = new Connections({
+        securityGroups: [attrs.securityGroup],
+        defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The default FSx file system type used by FSx for Lustre.
+   */
+  private static readonly DEFAULT_FILE_SYSTEM_TYPE: string = "LUSTRE";
+
+  /**
+   * The default port the file system listens on.
+   */
+  private static readonly DEFAULT_PORT: number = 988;
+
+  /**
+   * The security groups/rules used to allow network connections to the file system.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The ID that AWS assigns to the file system.
+   */
+  public readonly fileSystemId: string;
+
+  /**
+   * The encapsulated L1 file system.
+   */
+  private readonly fsxFileSystem: CfnFileSystem;
+
+  constructor(scope: Construct, id: string, props: FsxLustreFileSystemProps) {
+    super(scope, id);
+
+    this.validateProps(props);
+
+    const securityGroup = (props.securityGroup || new SecurityGroup(this, 'FsxLustreSecurityGroup', {
+      vpc: props.vpc
+    }));
+    securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port ingress');
+    securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port egress');
+
+    this.connections = new Connections({
+      securityGroups: Array.of(securityGroup),
+      defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+    });
+
+    this.fsxFileSystem = new CfnFileSystem(this, 'FsxLustreFileSystem', {
+      fileSystemType: FsxLustreFileSystem.DEFAULT_FILE_SYSTEM_TYPE,
+      subnetIds: Array.of(props.vpcSubnet.subnetId),
+      backupId: props.backupId,
+      kmsKeyId: (props.kmsKey ? props.kmsKey.keyId : undefined),
+      lustreConfiguration: props.lustreConfiguration,
+      securityGroupIds: Array.of(securityGroup.securityGroupId),
+      storageCapacity: props.storageCapacity
+    });
+
+    this.fileSystemId = this.fsxFileSystem.ref;
+    this.node.defaultChild = this.fsxFileSystem;
```

Comment:
This is not necessary if you name the construct `'Resource'`.

Reply:
Removed.